pkg/dockerclient/shell.go

```diff
@@ -288,14 +288,18 @@ func (c *ShellClient) RunContainer(imageName string, runOptions *RunOptions) (st
 				readonly = ",readonly"
 			}
 			dockerArguments = append(dockerArguments,
-				fmt.Sprintf("--mount %ssource=%s,destination=%s%s",
+				fmt.Sprintf("--mount '%ssource=%s,destination=%s%s'",
 					mountType,
 					mountPoint.Source,
 					mountPoint.Destination,
 					readonly))
 		}
 	}
 
+	for _, device := range runOptions.Devices {
+		dockerArguments = append(dockerArguments, fmt.Sprintf("--device '%s'", device))
+	}
```

Comment:
There is no validation on the `Devices` values. 
Aren't there any vulnerabilities here (injections etc)?

Reply:
using `common.Quote` which does exactly that